modules/ballerina-core/src/main/java/org/ballerinalang/model/values/BStruct.java

```diff
@@ -173,6 +180,150 @@ public void setRefField(int index, BRefType value) {
         refFields[index] = value;
     }
 
+    @Override
+    public void lockIntField(int index) {
+        if (longLocks == null || longLocks[index] == null) {
+            initIntLocks(index);
```

Comment:
Shall we divide init*Lock logic into two? Because inside the initIntlock there is another check for longLocks == null